addons/account/models/account_move.py

```diff
@@ -991,10 +991,13 @@ def _recompute_dynamic_lines(self, recompute_all_taxes=False, recompute_tax_base
         '''
         for invoice in self:
             # Dispatch lines and pre-compute some aggregated values like taxes.
+            needed_taxes = invoice.line_ids.tax_ids.flatten_taxes_hierarchy().filtered(lambda tax: tax.amount > 0)
```

Comment:
I haven't read the code around that but you may have to consider negative tax. (even if it's dumb)
```suggestion
            needed_taxes = invoice.line_ids.tax_ids.flatten_taxes_hierarchy().filtered(lambda tax: tax.amount != 0)
```

Reply:
With a float comparison

Reply:
> With a float comparison

Damn, why can't I learn the `is_zero()` lesson? :(

Reply:
I don't think we need to use `float_is_zero` here, this field is not calculated and it is directly entered by data files or human forms, so unless a government has a tax within order of 10^-15 (which is impossible because the governments are too greedy)  we are safe to go with normal comparison.

But to be on the safe (or sane?) side, I added a filter on the line balance, which is a monetary value, because I cannot easily determine the decimal precision of the tax amount since the amount holds both percent and monetary fixed value.